admin_ui/app/org/OrgStore.js

```diff
@@ -0,0 +1,21 @@
+import { BehaviorSubject } from 'rxjs';
+
+class OrgStore {
+  constructor() {
+    /*
+      In OrgEditPermissions Create, Update and Delete actions happen in modals.
+      The currentModal$ observable takes 4 possible values:
+      "" (empty string), "CREATE", "EDIT", "DELETE"
+     */
+    this.currentPermissionsModal$ = new BehaviorSubject('');
+    /*
+      Similarly for OrgEditRoles
+     */
+    this.currentRolesModal$ = new BehaviorSubject('');
+  }
+
```

Comment:
Question: Is it possible to have both a *role* modal and a *permission* modal open?

If not, then I suggest you merge the above observables and rename to simply `displayedModal$`. As for the values, it's always a great idea to introduce constants, i.e. create `admin_ui/app/constants/modalTypes.js` and add the following content...

```
export const PERMISSION_CREATE = 'PERMISSION_CREATE';
export const PERMISSION_EDIT = 'PERMISSION_EDIT';
// add more constants for permission and role
```

Then use these constants across the codebase, including the if statement in the react components above.

Reply:
Of course it's not possible to have both modals open!

Great suggestion this. implemented.